File: examples/bigclient.py
import requests
import json
import yaml
import sys
#Essential Variables
ipaddr = 'ip'
bigUrl = 'https://%s/mgmt' % ipaddr
user = 'username'
pwd = 'password'
#REQ Headers
headers = {
  'Content-Type' : 'application/json',
  'Accept'       : 'application/json'
}
#JSON Object Passed in POST REQ
loginData = {
  'username' : user,
  'password' : pwd
}

#Login to BigIQ
loginReq = requests.post(
  url=bigUrl + '/shared/authn/login',
  data=json.dumps(loginData),
  headers=headers,
  auth=(user, pwd),
  verify=False
)

#Check for Successful Login
if loginReq.status_code == 200:
  #Get Login Token Of Successful Login
  token = yaml.load(loginReq.text)['token']['token']
else:
  print 'Login was unsuccessful.'
  sys.exit(1)

#Add the Token to the Header
headers['X-F5-Auth-Token'] = token
#See the New Token Attached to the HEADER
#print headers

#Make Sample ECHO Call to BigIQ for Testing Token REQ
echoReq = requests.get(
  url=bigUrl + '/shared/echo',
  data=None,
  headers=headers,
  auth=(user, pwd),
  verify=False)

#Check for Successful Echo
if echoReq.status_code == 200:
  print yaml.load(echoReq.text)
else:
  print 'Echo RESP not Received.'

File: examples/bigclient.js
var request = require('request'),
    Q       = require('q');

var client;

function Client(options) {
  this.token = options.token || null;
  if (this.token) headers['X-F5-Auth-Token'] = this.token;
  this.reqOptions = {
    uri       : options.uri,
    method    : options.method,
    auth      : { user : options.user, pass : options.pass },
    header    : {
      'Content-Type' : 'application/json',
      'Accept'       : 'application/json'
    },
    strictSSL : false,
    json      : options.data
  };
};

Client.prototype._req = function() {
  var deferred = Q.defer();
  var resData = '';
  request(this.reqOptions)
    .on('data', function(data) {
      resData += data;
    })
    .on('end', function(res) {
      deferred.resolve(resData);
    })
    return deferred.promise;
};

exports.login = function(options) {
  var deferred = Q.defer();
  client = new Client(options);
  client._req()
    .then(function(data) {
      var login = JSON.parse(data);
      if (login.token.token) client.token = login.token.token;
      deferred.resolve(login.token.token);
    });
  return deferred.promise;
};

exports.echo = function(echo) {
  var deferred = Q.defer();
  client.reqOptions.json = echo.data;
  client.reqOptions.uri = echo.uri;
  client.reqOptions.method = echo.method;
  client._req()
    .then(function(resp) {
      deferred.resolve(resp);
    });
  return deferred.promise;
}
